### PR TITLE
Feat: added activity_log_retrieval

### PR DESCRIPTION
**Pull request by RichoKD**
Should close https://github.com/BlockheaderWeb3-Community/autoswappr-backend/issues/6

### CODE REVIEW

**Comment by Abeeujah** [comment]
Hi @RichoKD could you please fix failing CI

**Comment by RichoKD** [comment]
> Hi @RichoKD could you please fix failing CI

I've done it

File: src/api_error.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ impl IntoResponse for ApiError {
             ApiError::DatabaseError(ref err) => format!("{}", err),
             ApiError::InternalError(ref err) => format!("{}", err),
         };
-        error!("{}", error_to_log);
+        error!("ERROR -------> {}", error_to_log);
```
Comment (Abeeujah) [inline src/api_error.rs:49]: Hi, can you revert the change you made here?

File: tests/api/activity_log_retrieval.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        .unwrap();
+    let resp = app.request(req).await;
+    let headers = resp.headers().clone();
+    println!("{:#?}", resp.body());
+
+    assert_eq!(resp.status(), StatusCode::OK);
+    assert!(headers.get("x-request-id").is_some());
+    assert_eq!(headers.get("access-control-allow-origin").unwrap(), "*");
+    assert!(headers.get("vary").is_some());
+}
```
Comment (Abeeujah) [inline tests/api/activity_log_retrieval.rs:31]: Also could you include test cases that check for errors and edge cases eg empty transactions log

**Review by Abeeujah — CHANGES_REQUESTED** [review]

File: src/api_error.rs (hunk 1 above)
Comment (RichoKD) [reply src/api_error.rs:49]: Ok

File: tests/api/activity_log_retrieval.rs (hunk 2 above)
Comment (Abeeujah) [reply tests/api/activity_log_retrieval.rs:31]: _home_abee_Desktop_OdHack_autoswappr-backend_tarpaulin-report html
Here's the result of your current test coverage...

Comment (RichoKD) [reply tests/api/activity_log_retrieval.rs:102]: I've updated it

Comment (RichoKD) [reply tests/api/activity_log_retrieval.rs:152]: Done
